litebird_sim/beam_convolution.py

```diff
@@ -0,0 +1,313 @@
+                "you must do the same for `pointings`"
+            )
+            assert len(observations) == len(pointings), (
+                f"The list of observations has {len(observations)} elements, but "
+                + f"the list of pointings has {len(pointings)} elements"
+            )
+            obs_list = observations
+            ptg_list = pointings
+
+    for cur_obs, cur_ptg in zip(obs_list, ptg_list):
+        if isinstance(sky_alms, dict):
+            if all(item in sky_alms.keys() for item in cur_obs.name):
+                input_sky_names = cur_obs.name
+            elif all(item in sky_alms.keys() for item in cur_obs.channel):
+                input_sky_names = cur_obs.channel
+            else:
+                raise ValueError(
+                    "The dictionary maps does not contain all the relevant "
+                    "keys, please check the list of detectors and channels"
+                )
+            if "Coordinates" in sky_alms.keys():
+                dict_input_sky_alms_in_galactic = (
+                    sky_alms["Coordinates"] is CoordinateSystem.Galactic
+                )
+                if dict_input_sky_alms_in_galactic != input_sky_alms_in_galactic:
+                    logging.warning(
+                        "input_sky_alms_in_galactic variable in add_convolved_sky_to_observations"
+                        " overwritten!"
+                    )
+                input_sky_alms_in_galactic = dict_input_sky_alms_in_galactic
+        else:
+            assert isinstance(sky_alms, np.ndarray), (
+                "sky_alms must be either a dictionary contaning the keys for all the"
+                + "channels/detectors or a 3×N_ℓm NumPy array"
+            )
+            input_sky_names = None
+
+        if isinstance(beam_alms, dict):
+            if all(item in beam_alms.keys() for item in cur_obs.name):
+                input_beam_names = cur_obs.name
+            elif all(item in beam_alms.keys() for item in cur_obs.channel):
+                input_beam_names = cur_obs.channel
+            else:
+                raise ValueError(
+                    "The dictionary beams does not contain all the relevant "
+                    "keys, please check the list of detectors and channels"
+                )
+        else:
+            assert isinstance(beam_alms, np.ndarray), (
+                "beam_alms must be either a dictionary containing keys for all the "
+                "channels/detectors or a 3×N_ℓm NumPy array"
+            )
+            input_beam_names = None
+
+        if hwp is None:
+            if hasattr(cur_obs, "hwp_angle"):
+                hwp_angle = cur_obs.hwp_angle
+            else:
+                hwp_angle = None
+        else:
+            if type(cur_ptg) is np.ndarray:
+                hwp_angle = get_hwp_angle(cur_obs, hwp)
+            else:
+                logging.warning(
```

Comment:
In this `else` block, `hwp_angle` is not initialized and will cause an error in the call to `add_convolved_sky` later

Reply:
I think this should be an exception instead of a warning. What do you think, @paganol? If a pointing must be supplied and it isn't, there's no sense in continuing, correct?